src/cart/src/gurt/misc.c

```diff
@@ -107,7 +107,7 @@ d_rank_list_dup_sort_uniq(d_rank_list_t **dst, const d_rank_list_t *src)
 		D_GOTO(out, 0);
 	identical_num = 0;
 	rank_tmp = rank_list->rl_ranks[0];
-	for (i = 1; i < rank_num - identical_num; i++) {
+	for (i = 1; i < rank_num; i++) {
 		if (rank_tmp == rank_list->rl_ranks[i]) {
 			identical_num++;
 			for (j = i; j < rank_num; j++)
```

Comment:
Minor: This looks not efficient to me, but that's not in scope of this patch.

Reply:
Agreed, this needs to be reworked at some point, it's extremely confusing and was (without this change) returning duplicate entries.